hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -944,4 +944,12 @@
      */
     CacheStatistics getLocalCacheStatistics();
 
+    /**
+     * Opens cache if available (not destroyed).
+     * If cache is already open, does nothing.
+     *
+     * @throws java.lang.IllegalStateException if cache is already destroyed
+     */
+    void open();
```

Comment:
Via a new internal API like `ICacheInternal` (`ICache` will extends from this), we can handle some features via this interface. Reason for adding new method to public API is handling case from only `AbstractHazelcastCacheManager` instead of handling in both of `HazelcastClientCacheManager` and `HazelcastServerCacheManager` since there is no connection between `ClientCacheProxy` and `CacheProxy` except `ICache` interface.